src/sequence.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/**
+ * Turn any object into an iterator.
+ * Takes objects that implement the iterator protocol.
+ * Plain objects are treated as key-value stores and yield
+ * a sequence of their key value bytes, represented as size-2 arrays.
+ *
+ * Any value that is allowed as a parameter for this function shall be
+ * considered to be a `Sequence` for the purpose of this file.
+ * This term shall be distinguished from `Iterable` in that iterables
+ * must implement the iterator protocol `iterable[Symbol.iterator]()`.
+ *
+ * @param {Object|Iterable|Iterator} obj
+ * @returns {Iterator}
+ * @yields The data from the given elements
+ */
+const iter = obj => {
+  function *objIter() {
+    for (const key in obj) {
+      if (obj.hasOwnProperty(key)) {
+        yield [key, obj[key]];
+      }
+    }
+  }
+  return obj.constructor === Object ? objIter(obj) : obj[Symbol.iterator]();
```

Comment:
shouldn't you check if `Symbol.iterator` exists first, and fallback to `objIter()` ?

Reply:
Well, I don't have a strong opinion in this and that's how it is in the current version uploaded.
I tend to prefer not checking for Symbol.iterator, because I think when plain objects are used developers tend to think of them as associative arrays, so iterating over them should yield key
value pairs. By allowing custom Iterators we kind of break this expectation.

IMHO devs should define custom types (i.e. use a custom constructor even if it's NO-OP when they want a different behaviour).

Of course the real issue is javascript not distinguishing between Object (as a type base) and dict (as an associative array) ;)